src/github.com/couchbase/sync_gateway/base/sequence_clock.go

```diff
@@ -330,8 +330,8 @@ func (c *SyncSequenceClock) GetHashedValue() string {
 }
 
 func (c *SyncSequenceClock) SetHashedValue(value string) {
-	c.lock.RLock()
-	defer c.lock.RUnlock()
+	c.lock.Lock()
```

Comment:
can you add a comment on why a rw lock must be used rather than just a read lock?  (I think that might prevent a case where someone in the future changes it to a read lock as an optimization and then causes a bug)